OpenRA.Mods.Common/Traits/Buildings/Refinery.cs

```diff
@@ -102,14 +102,38 @@ public IEnumerable<TraitPair<Harvester>> GetLinkedHarvesters()
 
 		int IAcceptResources.AcceptResources(string resourceType, int count)
 		{
-			if (!playerResources.Info.ResourceValues.TryGetValue(resourceType, out var resourceValue))
+			var canAcceptRawResources = playerResources.CanAcceptSpecialRawResources(resourceType);
+			if (!playerResources.Info.ResourceValues.TryGetValue(resourceType, out var resourceValue) &&
+				!canAcceptRawResources)
 				return 0;
```

Comment:
Strange indention. Just put the if into one line to keep it readable.

Reply:
fixed